channel/player.go

```diff
@@ -240,16 +240,16 @@ func (d *player) setEXP(amount int32) {
 }
 
 func (d *player) giveEXP(amount int32, fromMob, fromParty bool) {
-	a := int32(d.rates.exp) * amount
+	amount = int32(math.Ceil(float64(d.rates.exp * float32(amount))))
```

Comment:
Ceil is not a cheap call to make especially as it gets called every time a players kills a mob or receives exp in general. Just do the standard type cast, this will round down but I think that's ok.

Also this will apply the exp rate to everything the player receives exp for e.g. both quests, PQ stage completion, mob kills. Might want to separate this out?

Reply:
If I remember correctly this is an expected behaviour from maple story